Add mkcert for certificates

arnested opened the pull request:
```sh
$ docker-compose logs web
Attaching to balder
balder     | *** Running /etc/my_init.d/00_regen_ssh_host_keys.sh...
balder     | *** Running /etc/my_init.d/10_syslog-ng.init...
balder     | Feb 15 10:16:59 d249936382ad syslog-ng[14]: syslog-ng starting up; version='3.13.2'
balder     | *** Running /etc/my_init.d/mkcert.sh...
balder     | Using the local CA at "/mkcert" ✨
balder     | The local CA is now installed in the system trust store! ⚡️
balder     |
balder     | Using the local CA at "/mkcert" ✨
balder     |
balder     | Created a new certificate valid for the following names 📜
balder     |  - "balder.docker"
balder     |  - "*.balder.docker"
balder     |  - "local.docker"
balder     |
balder     | Reminder: X.509 wildcards only go one level deep, so this won't match a.b.balder.docker ℹ️
balder     |
balder     | The certificate is at "/etc/ssl/certs/ssl-cert-snakeoil.pem" and the key at "/etc/ssl/private/ssl-cert-snakeoil.key" ✅
balder     |
balder     | *** Booting runit daemon...
balder     | *** Runit started as PID 733
balder     | Feb 15 10:17:02 d249936382ad cron[744]: (CRON) INFO (pidfile fd = 3)
balder     | Feb 15 10:17:02 d249936382ad cron[744]: (CRON) INFO (Running @reboot jobs)
```

arnested committed:
Add mkcert for certificates

arnested committed:
Add certificate to folder in format recognizable by Dory / nginx-ssl

arnested committed:
Locate CA root in away that is preconfigured to work on both Mac and Linux.

By searching for `rootCA.pem` and `rootCA-key.pem` in a folder beneath
`/mkcert` instead of assuming hardcoded locations we can make different
volumes for Mac and Linux regardless of whether we actually run Mac or
Linux.

danquah commented:
This works for me on dory. Do need some where to document how ones docker-compose setup should be set up for this to work ... should should we just do it in this repo? In which case I'll just go ahead and update the readme

danquah commented on README.md:
```diff
@@ -1,3 +1,33 @@
+      MKCERT_DOMAINS: "example.docker *.example.docker local.docker"
+      VIRTUAL_HOST: example.docker
+
+    hostname: example
```
The documentation I could find on hostname and domainname was sketchy at best. What is the usecase for this @arnested ?

arnested replied:
https://docs.docker.com/compose/compose-file/#domainname-hostname-ipc-mac_address-privileged-read_only-shm_size-stdin_open-tty-user-working_dir

It sets the host name and domain name for the container.

The `mkcert.sh` in the container will if neither `MKCERT_DOMAINS` or `VIRTUAL_HOST` environment variables are set, use the output of `hostname -f` (which can be set via those options).

danquah committed:
Add documentation on how to integrate the mkcert support with Dory

arnested committed:
Create expected foldes in Dockerfile

Create expected foldes in Dockerfile so they will be created while
building the image instead of doing it when running the script (`/cert`)
and not at all (`/mkcert`).

arnested committed:
Improve documentation on mkcert part

arnested committed:
Support Dinghy wildcard domains

Dinghys wildcard syntax is just prefix with a dot (`.example.com`) where
as mkcert uses an asterisk (`*.example.com`).

So we rewrite the `VIRTUAL_HOST` if it starts with a dot.

To do that we eed to split up the compound onliner finding the domains
in either `MKCERT_DOMAINS`, `VIRTUAL_HOST` or hostname. But that is
probably better for reading and understanding anyway (at least it is
easier to document all steps now).

arnested committed:
Add more documentation to mkcert.sh script

arnested committed:
Be verbose about copying into /cert

Make sure we are verbose in the log about which files we copy into
/cert.
